tests/test_driver.py

```diff
@@ -342,3 +343,11 @@ def test_ami_product_release(mock_get_client):
 def test_get_entity_versions(mock_get_details):
     mock_get_details.return_value = {"Versions": [{"CreationDate": "20231010"}, {"CreationDate": "20230202"}]}
     assert _driver.get_entity_versions("foo") == [{"CreationDate": "20230202"}, {"CreationDate": "20231010"}]
+
+
+@patch("awsmp._driver.get_client")
+def test_get_no_offer_id(mock_get_client):
```

Comment:
@rthill91 I tried to add test with raising ClientError from botocore but is there any better way to raise them? 

I added ClientError as side_effect, then I needed to pass [two arguments](https://github.com/boto/botocore/blob/dcadf474a9e713fdf6fc3c7b174d689a5b160663/botocore/exceptions.py#L520) which one of them is detailed error messages and dictionary but I think it has better way to do it?